src/tests/basic/turn.ts

```diff
@@ -0,0 +1,227 @@
+        P.hasBuiltWorkerThisTurn = true;
+
+        techCards($, []);
+
+        expect(P.hasShuffledThisTurn).to.equal(false);
+        expect(P.hasBuiltWorkerThisTurn).to.equal(false);
+      });
+
+      it("should ready all in-play cards for the player", () => {
+        const fatigued = makeInstance($, $.activePlayer, "Nautical Dog");
+        const exhausted = makeInstance($, $.activePlayer, "Nautical Dog");
+        const disabled = makeInstance($, $.activePlayer, "Nautical Dog");
+        const patrolling = makeInstance($, $.activePlayer, "Nautical Dog");
+
+        exhausted.arrivalFatigue = false;
+        disabled.arrivalFatigue = false;
+        patrolling.arrivalFatigue = false;
+
+        exhausted.readyState = "EXHAUSTED";
+        disabled.readyState = "DISABLED";
+
+        P.patrol.squadLeader = patrolling.id;
+
+        techCards($, []);
+
+        expect(fatigued.arrivalFatigue).to.equal(false);
+        expect(exhausted.arrivalFatigue).to.equal(false);
+        expect(disabled.arrivalFatigue).to.equal(false);
+        expect(patrolling.arrivalFatigue).to.equal(false);
+
+        expect(fatigued.readyState).to.equal("READY");
+        expect(exhausted.readyState).to.equal("READY");
+        expect(disabled.readyState).to.equal("EXHAUSTED");
+        expect(patrolling.readyState).to.equal("READY");
+
+        expect(P.patrol.squadLeader).to.equal(null);
+      });
+    });
+
+    describe("Upkeep Phase", () => {
+      it("should give player a gold for every worker", () => {
+        techCards($, []);
+
+        expect(P.gold).to.equal(P.workers);
+      });
+
+      it(`should cap the player's gold at ${MAX_GOLD}`, () => {
+        P.gold = MAX_GOLD - 1;
+
+        techCards($, []);
+
+        expect(P.gold).to.equal(MAX_GOLD);
+      });
+    });
+
+    describe("Draw Phase", () => {
+      beforeEach(() => {
+        $ = initDummyGameState();
+        $.turnPhase = "MAIN";
+
+        P = $.players[$.activePlayer];
+
+        P.hand = [
+          "Nautical Dog",
+          "Mad Man",
+          "Bombaster",
+          "Careless Musketeer",
+          "Bloodrage Ogre",
+        ];
+
+        P.deck = [
+          "Makeshift Rambaster",
+          "Bloodburn",
+          "Scorch",
+          "Charge",
+          "Pillage",
+        ];
+      });
+
+      it(
+        "should refill the player's hand by 2",
+        () => {
+          P.hand = P.hand.slice(0, 2);
+
+          endTurn($);
+
+          expect(P.hand.length).to.equal(4);
+        }
+      );
+
+      it(
+        `should not let the player's hand exceed ${MAX_HAND_SIZE} cards`,
+        () => {
+          endTurn($);
+
+          expect(P.hand.length).to.equal(5);
+        }
+      );
+
+      it("should replace the player's current hand", () => {
+        endTurn($);
+
+        expect(P.hand.includes("Nautical Dog")).to.equal(false);
+      });
+    });
+
+    describe("Tech Phase", () => {
+      it("should allow not teching cards on the first round", () => {
```

Comment:
This test seems weird, since in ordinary play it should be impossible to be in TECH phase on round 1. Maybe a more appropriate test is "tech phase is skipped on first round"

Reply:
I'll adjust the wording.